src/tdastro/sources/bayeSN_model.py

```diff
@@ -0,0 +1,493 @@
+        """Compute a mask for whether a given time is of interest for a given object.
+        For example, a user can use this function to generate a mask to include
+        only the observations of interest for a window around the supernova.
+
+        Parameters
+        ----------
+        times : numpy.ndarray
+            A length T array of observer frame timestamps in MJD.
+        graph_state : GraphState, optional
+            An object mapping graph parameters to their values.
+
+        Returns
+        -------
+        time_mask : numpy.ndarray
+            A length T array of Booleans indicating whether the time is of interest.
+        """
+        if graph_state is None:
+            raise ValueError("graph_state needed to compute mask_by_time")
+
+        z = self.get_param(graph_state, "redshift", 0.0)
+        if z is None:
+            z = 0.0
+
+        t0 = self.get_param(graph_state, "t0", 0.0)
+        if t0 is None:
+            t0 = 0.0
+
+        # Compute the mask.
+        good_times = (times > t0 + -20.0 * (1.0 + z)) & (times < t0 + 50.0 * (1.0 + z))
+        return good_times
+
+    # MAIN FUNCTION:
+    def compute_flux(self, times, wavelength, graph_state, **kwargs):
+        """Draw effect-free observations for this object.
+
+        Parameters
+        ----------
+        times : numpy.ndarray
+            A length T array of rest frame timestamps in MJD.
+        wavelengths : numpy.ndarray, optional
+            A length N array of rest frame wavelengths (in angstroms).
+        graph_state : GraphState
+            An object mapping graph parameters to their values.
+
+        **kwargs : dict, optional
+            Any additional keyword arguments.
+
+        Returns
+        -------
+        flux_density : numpy.ndarray
+            A length T x N matrix of SED values (in nJy).
+        """
+
+        W_0 = self._W0_
+        W_1 = self._W1_
+        l_knots = self._l_knots_
+        tau_knots = self._tau_knots_
+        hsiao_wave = self._hsiao_wave
+        hsiao_phase = self._hsiao_phase
+        hsiao_flux = self._hsiao_flux
+        params = self.get_local_params(graph_state)
+
+        l = wavelength
+        tau = times - params["t0"]
+        # l = np.linspace(3000, 8000, 200)
+        # tau = np.linspace(-20, 50, 71)
+        W = W_0 + params["theta"] * W_1
+        invKD_l = self.compute_invKD(l_knots)
+        J_l = self.natural_cubic_spline_basis_matrix_from_invKD(l_knots, l, invKD_l)
+        invKD_tau = self.compute_invKD(tau_knots)
+        J_t = self.natural_cubic_spline_basis_matrix_from_invKD(tau_knots, tau, invKD_tau)
+        J_t_T = np.atleast_2d(J_t).T
+        WJt = np.matmul(W, J_t_T)
+        W_grid = np.matmul(J_l, WJt)
+        W_grid = np.atleast_2d(W_grid).T
+        H_grid = self.evaluate_2d_cubic_spline(hsiao_phase, hsiao_wave, hsiao_flux, tau, l)
+        flux_density = H_grid * 10 ** (-0.4 * W_grid)
+
+        # Apply dust extinction law
```

Comment:
This has the potential to cause some weirdness if someone unfamiliar with the model tries to add a dust extinction effect later.  Let me think of a different way to do this and get back to you.

Reply:
We can keep it as-is for now and I can change it later (after addressing #365 )

Reply:
@mi-dai pointed out that this is for the host galaxy extinction.  So this computation makes sense here.